_posts/2018-03-07-foreman-117-fact-filtering.md

```diff
@@ -91,3 +91,8 @@ be used as well):
 
     records = FactName.where(query).delete_all
     puts "Removed #{records} fact names"
+
+Please note that the settings for "Exclude pattern for facts stored in Foreman" and "Ignore
+interfaces with matching identifier" have some additional logic to simplify regex writing. For
+example, if "exclude pattern for facts" contains `foo01`, it will match `interfaces::foo01::mac`.
+Typically you will not want to copy/paste from one field to another.
```

Comment:
Actually it tries to equalize the semantics for both settings: if you want to get rid of all interfaces that start with `veth`, it means that you also don't want facts that are under `veth` in facts subtree (for hierarchical facts). The regex that is generated under the hood, tries to match common hierarchy delimiters to use as start and finish for fact name matching: we know that RHSM facts are using `::` do delimit fact names, so the regex will be "starts with ::, contains the string `veth` and delimited by ::"; we also know that puppet delimits its hierarchy with `_`, so for that sort of facts, we will add "satrts with _, contains the string `veth` and delimited by _" regex to the search.
Both cases express user's intent to remove `veth*` facts, so I do think that copy and paste is acceptable here. I would say that facts filter should be a superset of the interfaces filter, since the user will want additional facts be filtered out, like blockdevices e.t.c.

Reply:
The issue I hit with copy/paste was `*veth*`, which is allowed by "exclude pattern" but not "ignore interfaces".

Reply:
Ah, OK.